### PR TITLE
console: respect read-only mode in actions pages (close #4656)

### LINKED ISSUE #4656: console: make actions respect read-only mode
Currently, actions and custom types do not respect read-only mode.

### PR DESCRIPTION
**Pull request by soorajshankar**
### Description

Actions in Hasura  console will now respond to read only mode. This will hide/disable components when read only mode is enabled in the global state.

### Changelog

- [x] `CHANGELOG.md` is updated with user-facing content relevant to this PR.

### Affected components

- [x] Console

### Related Issues

close #4656

### Solution and Design

Referring to Data and Remote Schema, read only mode is implemented. 
Solution will 

- Hide components if the action is just a button or navigation components (Edit, Create, Remove Buttons)
- When there an action components which also shows information, component will be disabled but the information will still show up. (Editors, text area, input etc.)

### Steps to test and verify

1. Go to Hasura Console in read only mode (for testing you may have to set it in the [main](https://github.com/hasura/graphql-engine/blob/ff3ac78d8f11692b939ebb7a221ea1ae73930d4c/console/src/components/Main/State.js#L5) redux state )
2. Go to Actions
3. Create Table buttons from both sidebar and landing page should not be visible.
4. Select any existing actions (create without read only mode if there is nothing)
5. Modify
   - All inputs should be disabled.
   - Buttons should be hidden.
6. Relationships
   - Edit & Add Relationships button should be disabled.
&. Permissions
   - No Create/Edit operations should work (Similar to Data module)

### CODE REVIEW

File: console/src/components/Services/Actions/Landing/Main.js [hunk 1]
```diff
@@ -36,7 +39,7 @@ class Landing extends React.Component {
         dispatch(push(`${globals.urlPrefix}${appPrefix}/manage/add`));
       };
 
-      const addBtn = (
+      const addBtn = showActionBtns && (
```
Comment (wawhal) [inline console/src/components/Services/Actions/Landing/Main.js:42]: Can directly use `!readOnlyMode`?

File: console/src/components/Services/Actions/Modify/ActionEditor.js [hunk 2]
```diff
@@ -127,37 +132,43 @@ const ActionEditor = ({
...
-        <Button
-          color="red"
-          size="sm"
-          type="submit"
-          onClick={onDelete}
-          disabled={isFetching}
-        >
-          Delete
-        </Button>
+        {!readOnlyMode && (
```
Comment (wawhal) [inline console/src/components/Services/Actions/Modify/ActionEditor.js:149]: Can do the check only once for both:

```js
!readOnlyMode && (
  <React.Fragment>
    <Button>Save</Button>
    <Button>Delete</Button>
  </React.Fagment>
)
```

File: console/src/components/Services/Actions/Modify/ActionEditor.js [hunk 3]
```diff
@@ -79,6 +80,7 @@ const ActionEditor = ({
   };
 
   const allowSave =
+    !readOnlyMode &&
```
Comment (wawhal) [inline console/src/components/Services/Actions/Modify/ActionEditor.js:83]: Unnecessary since we are hiding the buttons anyway.

Comment (soorajshankar) [reply console/src/components/Services/Actions/Modify/ActionEditor.js]: @wawhal Done

File: console/src/components/Services/Actions/Modify/ActionEditor.js (hunk 2 above)
Comment (soorajshankar) [reply console/src/components/Services/Actions/Modify/ActionEditor.js:148]: Done

File: console/src/components/Services/Actions/Landing/Main.js (hunk 1 above)
Comment (soorajshankar) [reply console/src/components/Services/Actions/Landing/Main.js]: Done

File: console/src/components/Common/Layout/ExpandableEditor/Editor.js [hunk 4]
```diff
@@ -29,7 +29,17 @@ class Editor extends React.Component {
...
+      readOnlyMode = false,
+      isCollapsable,
+      service,
+      property,
+      collapseButtonText,
+      expandButtonText,
+    } = this.props;
+
+    const { isEditing } = this.state;
+    if (isCollapsable === false && isEditing) {
```
Comment (beerose) [inline console/src/components/Common/Layout/ExpandableEditor/Editor.js:42]:
```suggestion
    if (isCollapsable && isEditing) {
```

File: console/src/components/Services/Actions/Common/components/HeaderConfEditor.js [hunk 5]
```diff
@@ -8,15 +8,16 @@ const editorTooltip =
...
   className,
   forwardClientHeaders,
   toggleForwardClientHeaders,
   headers,
   setHeaders,
+  disabled = false,
 }) => {
   const getHeaders = () => {
-    return <Headers headers={headers} setHeaders={setHeaders} />;
+    return <Headers {...{ headers, setHeaders, disabled }} />;
```
Comment (beerose) [inline console/src/components/Services/Actions/Common/components/HeaderConfEditor.js:20]:
```suggestion
    return <Headers headers={headers} setHeaders={setHeaders} disabled={disabled} }} />;
```

File: console/src/components/Services/Actions/Landing/Main.js [hunk 6]
```diff
@@ -13,6 +13,7 @@ const actionsArchDiagram = `${globals.assetsPath}/common/img/actions.png`;
 class Landing extends React.Component {
   render() {
     const styles = require('../Actions.scss');
```
Comment (beerose) [inline console/src/components/Services/Actions/Landing/Main.js:15]: As you're here, could you move to the top and use import?

File: console/src/components/Services/Actions/Sidebar/LeftSidebar.js [hunk 7]
```diff
@@ -4,8 +4,13 @@ import { Link } from 'react-router';
...
 import styles from '../../../Common/Layout/LeftSubSidebar/LeftSubSidebar.scss';
 
-const LeftSidebar = ({ appPrefix, common: { actions, currentAction } }) => {
+const LeftSidebar = ({
+  appPrefix,
+  common: { actions, currentAction },
+  readOnlyMode,
+}) => {
   const [searchText, setSearchText] = React.useState('');
+  const showActionBtns = !readOnlyMode;
```
Comment (beerose) [inline console/src/components/Services/Actions/Sidebar/LeftSidebar.js:13]: Used once, can be inlined

File: console/src/components/Services/Actions/Permissions/Permissions.js [hunk 8]
```diff
@@ -119,7 +120,7 @@ const Permissions = ({
             permType: queryType,
             className,
             editIcon: editIcon,
-            onClick,
+            onClick: !readOnlyMode && onClick,
```
Comment (beerose) [inline console/src/components/Services/Actions/Permissions/Permissions.js:123]: It results in an error:

Screenshot 2020-06-03 at 16 31 17

File: console/src/components/Common/Layout/ExpandableEditor/Editor.js (hunk 4 above)
Comment (soorajshankar) [reply console/src/components/Common/Layout/ExpandableEditor/Editor.js:42]: this condition was already there in the previous code, do you want to toggle `isCollapsable` check and return null if it is true?

Comment (soorajshankar) [reply console/src/components/Common/Layout/ExpandableEditor/Editor.js:42]: or did you mean `if (!isCollapsable && isEditing) {`

Comment (beerose) [reply console/src/components/Common/Layout/ExpandableEditor/Editor.js:42]: Yes, this is what I meant. Updated

**Pull request merged**